### PR TITLE
glab: fix non-bash shell completion

### PR DESCRIPTION
**Pull request by Josh-Tilles**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Because `glab` expects the `-s`/`--shell` option, the formula mistakenly installed the same bash completion for both zsh and fish as well.

### CODE REVIEW

**Comment by MerelyAPseudonym** [comment]
(related: https://github.com/Homebrew/homebrew-core/pull/71574)

**Comment by MerelyAPseudonym** [comment]
To illustrate the effect of the change:

Before the fix:
```sh-session
$ cd "$(brew --prefix glab)"
$ shasum etc/bash_completion.d/glab share/fish/vendor_completions.d/glab.fish share/zsh/site-functions/_glab
a8dae08411f502730460eaee1a9fd741460938b9  etc/bash_completion.d/glab
a8dae08411f502730460eaee1a9fd741460938b9  share/fish/vendor_completions.d/glab.fish
a8dae08411f502730460eaee1a9fd741460938b9  share/zsh/site-functions/_glab
```

After the fix:
```sh-session
$ head -n 1 "$(brew --prefix glab)/share/zsh/site-functions/_glab"
#compdef _glab glab

# zsh completion for glab                                 -*- shell-script -*-
$ head -n 1 "$(brew --prefix glab)/share/fish/vendor_completions.d/glab.fish"
# fish completion for glab
```

**Review by Rylan12 — DISMISSED** [review]
Thanks, @MerelyAPseudonym!

Can you add a `revision 1` line so all users will have their `glab` installation updated with the correct completions?

**Comment by MerelyAPseudonym** [comment]
> Thanks, @MerelyAPseudonym!
> 
> 
> 
> Can you add a `revision 1` line so all users will have their `glab` installation updated with the correct completions?

Oops! Will do.

facepalm

**Comment by MerelyAPseudonym** [comment]
> Thanks, @MerelyAPseudonym!
> 
> 
> 
> Can you add a `revision 1` line so all users will have their `glab` installation updated with the correct completions?

Coming right up. Thanks for the reminder.

**Comment by MerelyAPseudonym** [comment]
@Rylan12 I rebased to include the revision bump in the single commit. If that was incorrect, just let me know and I can split it up into two commits.

**Comment by MerelyAPseudonym** [comment]
Actually… it might be worth examining `glab`’s history to see if introducing `--shell` means that manipulating the `SHELL` environment isn’t necessary anymore. It might be an opportunity to simplify the formula a little bit. I could look into it tomorrow.

I’ll leave it to the maintainers to decide whether the potential cleanup is worth delaying this PR or whether it should go in a subsequent PR.

**Comment by Rylan12** [comment]
Oh yeah, good call. Those can be removed, I think.

**Comment by MerelyAPseudonym** [comment]
Well, @Rylan12, I've reached the “I don’t think this _ever_ worked” stage with the formula’s current method of using `safe_popen_read` and setting `$SHELL`. As far as I can tell, [it was always necessary](https://github.com/profclems/glab/pull/215/files#diff-e30a7f65f143b1b001d5327556c7b0e2a087c789d3c5427dd0ee189100a21306R25) to specify the shell with an option rather than an environment variable.

Caveat: I am a dilettante when it comes to the Go ecosystem.

File: Formula/glab.rb
Comment (Rylan12) [inline Formula/glab.rb:24]:
```suggestion
    (bash_completion/"glab").write Utils.safe_popen_read(bin/"glab", "completion", "--shell=bash")
    (zsh_completion/"_glab").write Utils.safe_popen_read(bin/"glab", "completion", "--shell=zsh")
    (fish_completion/"glab.fish").write Utils.safe_popen_read(bin/"glab", "completion", "--shell=fish")
```

Yeah, it's totally possible that it never worked.

Almost there. I'd like to keep using `Utils.safe_popen_read` rather than the backticks. That way, if the command changes in the future, an error will be thrown letting us know.

Comment (MerelyAPseudonym) [reply Formula/glab.rb:24]: Makes sense to me!

For (my) future reference: should I _always_ prefer `Utils.safe_popen_read` over backticks when writing new code in Homebrew? And, if so, is there an official Homebrew backlog of desired but not yet implemented audit checks that we could add that rule to?

Comment (MerelyAPseudonym) [reply Formula/glab.rb]: (unresolving the conversation solely because of my pending question)

**Comment by MerelyAPseudonym** [comment]
squashed back into one commit, but kept the coauthor line GitHub added because why not 🙂

Also, I’ll hoist my question for @Rylan12 out from the resolved conversation:

For (my) future reference: should I _always_ prefer `Utils.safe_popen_read` over backticks when writing new code in Homebrew? And, if so, is there an official Homebrew backlog of desired-but-not-yet-implemented audit checks that we could add that rule to?

**Review by Rylan12 — APPROVED** [review]
Thanks, @MerelyAPseudonym!

In general, I think we do prefer `Utils.safe_popen_read` (or `Utils.popen_read` in some cases) to backticks, but it's possible that there are a few exceptions. I'm not sure. You could take a look at adding an audit for that if you'd like. You can check out the existing rule that requires using `Utils.safe_popen_read` over `Utils.popen_read` in some cases for an example of this type of style check. That code is [here](https://github.com/Homebrew/brew/blob/f14ed4987f46de2dff36e0bc0b40615d56fa3c49/Library/Homebrew/rubocops/lines.rb#L215-L242).

**Pull request closed without merging**
